src/threading/thread.cpp

```diff
@@ -177,6 +180,8 @@ void Thread::threadProc(Thread *thr)
 	thr->m_kernel_thread_id = thread_self();
 #endif
 
+	current_thread = thr;
+
 	thr->setName(thr->m_name);
 
 	g_logger.registerThread(thr->m_name);
```

Comment:
looks like the logger could possibly use the new method too :)

Reply:
The logger also handles non-registered threads:

https://github.com/minetest/minetest/blob/cdce33dd05cfc9c13900e5a7cffcc41f8051237b/src/log.cpp#L236-L248

So would it be safe to assume that `Thread::getCurrentThread() == nullptr` means "main thread"? Or are there threads not handled by the `Thread` class?

Reply:
Not sure. But this also doesn't matter too much, better leave it as-is.